### PR TITLE
Several improvements based on issues

### PR DESCRIPTION
**Pull request by klein0r**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1270,18 +1304,32 @@ function checkCode(context) {
...
+            readFiles.push('admin/index_m.html');
+            readFiles.push('admin/words.js');
+        }
+
+        if (context.ioPackageJson.common.adminUI && context.ioPackageJson.common.adminUI.config === 'json') {
+            readFiles.push('admin/jsonConfig.json');
+            readFiles.push('admin/i18n/en/translations.json');
+        }
+
+        if (context.ioPackageJson.common.jsonCustom || (context.ioPackageJson.common.adminUI && context.ioPackageJson.common.adminUI.custom === 'json')) {
```
Comment (Apollon77) [inline index.js:1325]: Maybe only check for "supportsCustom" because admin5 requires (hard) to have jsonConfig for customs

File: index.js [hunk 2]
```diff
@@ -1330,59 +1378,99 @@ function checkCode(context) {
...
+                        } catch (e) {
+                            context.errors.push('[E511] Cannot parse "admin/jsonCustom.json": ' + e);
+                        }
+                    } else {
+                        context.errors.push(`[E512] "admin/jsonCustom.json" not found, but custom support is declared`);
                     }
                 }
 
+                if (context['/iob_npm.done']) {
+                    context.errors.push('[E503] "iob_npm.done" found in repo! Please add it to .gitignore');
```
Comment (Apollon77) [inline index.js:1444]: Better "please remove this file"

**Review by Apollon77 — CHANGES_REQUESTED** [review]
Thank you very much for the PR, some comments

File: index.js (hunk 2 above)
Comment (klein0r) [reply index.js:1444]: I have not idea what that file is used for (that rule already existed before this PR)

Comment (Apollon77) [reply index.js:1444]: When you install an adapter in your iobroker system then iobroker creates that file in the dir after installation ... or better "Older versions of controller". And some developers dvelop the adapter directly in iorboekr and then copy all stuff over ... this means that they also copy that file ... thats why we check it and tell it is deleted

File: index.js [hunk 3]
```diff
@@ -943,8 +963,7 @@ function checkIOPackageJson(context) {
                                 context.errors.push(`[E153] common.dependencies must contain {"js-controller": ">=1.5.8"} or {"js-controller": ">=2.0.0"} or {"js-controller": ">=3.0.0"}`);
                             }
                         }
-                    } else
-                    if (context.packageJson.dependencies['@iobroker/adapter-core'].includes('2.4') || context.packageJson.dependencies['@iobroker/adapter-core'].includes('2.5')) {
+                    } else if (context.packageJson.dependencies['@iobroker/adapter-core'].includes('2.4') || context.packageJson.dependencies['@iobroker/adapter-core'].includes('2.5')) {
```
[the remark this answers is not in the document]
Comment (klein0r) [reply index.js:966]: Can you provide a list of checks? I don't know why there are so complex rules to check.

[the remark this answers is not in the document]
Comment (Apollon77) [reply index.js:966]: in fact it is:
* adapter core <2.3 has no special dep requirement
* adapter-core 2.3 requires js-controller 1.5.8+ as dep
* adapter-core 2.4+ required js-controller 2.0.0+ as dep

**Comment by klein0r** [comment]
@Apollon77 Review please

**Comment by GermanBluefox** [comment]
Thank you!

**Comment by GermanBluefox** [comment]
There is a new possibility, how to start a repochecker:
```
npx @iobroker/repochecker https://github.com/ioBroker/ioBroker.javascript master
```

**Comment by Apollon77** [comment]
I cxreate docs und portal issues

**Pull request merged**